main.go

```diff
@@ -35,10 +34,10 @@ func checkForErrorsAndExit(err error) {
 			logger.Println(err)
 		}
 		// exit with the underlying error code
-		var retCode int = 1
-		if exiterr, ok := errors.Unwrap(err).(*exec.ExitError); ok {
-			status := exiterr.Sys().(syscall.WaitStatus)
-			retCode = status.ExitStatus()
+		var retCode int
+		retCode, err := shell.GetExitCode(err)
```

Comment:
It's very confusing here to shadow both the `retCode` and `err` variables. I'd recommend removing the extra `retCode` declaration and renaming this `err` to `exitCodeErr`.